cocos/editor-support/cocostudio/ActionTimeline/CCBoneNode.h

```diff
@@ -0,0 +1,197 @@
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+****************************************************************************/
+
+
+#ifndef __CCBONENODE_H__
+#define __CCBONENODE_H__
+
+#include "base/CCProtocols.h"
+#include "2d/CCLayer.h"
+#include "CCTimelineMacro.h"
+#include "cocostudio/CocosStudioExport.h"
+
+#include "CCSkinNode.h"
+
+NS_TIMELINE_BEGIN
+
+class SkeletonNode;
+
+class CC_STUDIO_DLL BoneNode : public cocos2d::Node, public cocos2d::BlendProtocol
+{
+public:
+    static BoneNode* create();
+    static BoneNode* create(const int &length);
+
+    using Node::addChild;
+    // add child, deal with BoneNode and SkinNode
+    virtual void addChild(cocos2d::Node* child, int localZOrder, const std::string &name) override;
+    virtual void addChild(cocos2d::Node* child, int localZOrder, int tag) override;
+
+    virtual void removeChild(Node* child, bool cleanup) override;
+
+    virtual const  cocos2d::Vector<BoneNode*>& getChildBones() const { return _childBones; }
+    virtual cocos2d::Vector<BoneNode*>&  getChildBones() { return _childBones; }
+
+    /**
+    *@brief: add a skin
+    *@param: display, whether display this skin
+    */
+    virtual void addSkin(SkinNode* skin, bool display);
+
+    /**
+    *@brief: add a skin
+    *@param: display, whether display this skin
+    *@param: hideOthers, whether hide other skins added to this bone
+    */
+    virtual void addSkin(SkinNode* skin, bool display, bool hideOthers);
+
+    virtual const  cocos2d::Vector<SkinNode*>& getSkins() const { return _boneSkins; }
+    virtual  cocos2d::Vector<SkinNode*>&  getSkins() { return _boneSkins; }
+
+    /**
+    * @brief: display skin
+    * @param: hideOthers, set other skins invisible 
+    */
+    virtual void displaySkin(SkinNode* skin, bool hideOthers);
+
+    /**
+    * @brief: display all skins named skinName, if hide display only one skin,
+    *         prefer to use display(SkinNode* skin, bool hideOthers = false)
+    * @param: hideOthers, set other skins invisible
+    */
+    virtual void displaySkin(const std::string &skinName, bool hideOthers );
+
+    // get the skins which is visible (displaying skins)
+    virtual cocos2d::Vector<SkinNode*> getVisibleSkins() const;
+    
+    virtual SkeletonNode* getRootSkeletonNode() const;
```

Comment:
please re-check all the doxygen comments for all public API. It's very important, because we extract these information to generate API reference in api.cocos.com